client/src/components/NationwideMap.vue

```diff
@@ -336,55 +339,59 @@ export default defineComponent({
 
       this.setUpInteractionHandlers();
       this.setUpControls();
-      this.setUpZoomListener();
+      this.map?.on('zoom', this.toggleDataOnZoom);
 
       // If the map has nothing on it, check the current data layer.
       if (this.visibleLayer == null && this.currentDataLayerId != null) {
         this.updateMapOnDataLayerChange(ALL_DATA_LAYERS.get(this.currentDataLayerId));
       }
-
     },
 
     /**
      * Creates base Mapbox map and configures interactions and styles for
      * layers.
      */
     async createMap(): Promise<void> {
+      // Start zoomed in for a scorecard to avoid unnecessary tile loads.
+      const zoom = this.scorecard ? PARCEL_ZOOM_LEVEL : DEFAULT_ZOOM_LEVEL;
```

Comment:
nit: will this zoom out if we don't have parcel info? i.e. if we return just water system info, will this zoom out to zip level? if so we should make that default since that will be most cases.

Reply:
I didn't see a good ZIP zoom level to use, since the zoom depends on the size of the shape that the latlong points to. I added some TODOs to further optimize this.